IdentityServer/Startup.cs

```diff
@@ -82,14 +95,58 @@ public void ConfigureServices(IServiceCollection services)
+                    {
+                        options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                        options.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                        options.DefaultChallengeScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                    })
+                    // Adds Fontys Single Sign On authentication.
+                    .AddOpenIdConnect("oidc", "Fontys", options =>
+                        {
+                            options.ClientId = Config.FfhictOIDC.ClientId;
+                            options.ClientSecret = Config.FfhictOIDC.ClientSecret;
+                            options.Authority = Config.FfhictOIDC.Authority;
+                            options.ResponseType = "code";
+                            options.Scope.Clear();
+
+                            string[] scopes = Config.FfhictOIDC.Scopes.Split(" ");
+                            foreach(string scope in scopes)
+                            {
+                                options.Scope.Add(scope);
+                            }
+
+                            // Set this flow to get the refresh token.
+                            // options.Scope.Add("offline_access");
+
+                            options.SaveTokens = true;
+                            options.GetClaimsFromUserInfoEndpoint = true;
+
+                            // This sets the redirect uri, this is needed because the blackbox implementation does not implement fontys SSO.
+                            options.Events.OnRedirectToIdentityProvider = async n =>
+                            {
+                                n.ProtocolMessage.RedirectUri = Config.FfhictOIDC.RedirectUri;
+                                await Task.FromResult(0);
+                            };
+                        }
+                        // Add jwt validation this is so that the DGS can authenticate.
+                    ).AddJwtBearer(o =>
+                    {
+                        o.SaveToken = true;
+                        o.Authority = Config.Self.JwtAuthority;
+                        o.RequireHttpsMetadata = false;
+                        o.TokenValidationParameters = new TokenValidationParameters()
+                                                      {
+                                                          ValidateActor = false,
+                                                          ValidateAudience = false,
+                                                          NameClaimType = "name",
+                                                          RoleClaimType = "http://schemas.microsoft.com/ws/2008/06/identity/claims/role"
+                                                      };
+                    })
+                    .AddCookie();
 
             //TODO: Have some sort of certificate on the production servers
```

Comment:
We're moving into production, should this be changed now?

Reply:
This is already fixed in my branch, I will create a PR for this later today (Since this is a task of infrastructure).